src/nodes.rs

```diff
@@ -73,30 +73,41 @@ pub fn get_cell_position(
 }
 
 pub async fn create_nodes(
-    agraph: &AGraph,
+    indices: &Vec<Vec<usize>>,
     node_ips: &[String],
     geo_cache: &GeoIPCache,
 ) -> Vec<Node> {
-    let graph: Graph<usize> = Graph::new();
-    // TODO(asmie/kylegranger): make this an associated function for Graph.
-    // it does not use the Graph per se
-    let (betweenness, closeness) = graph.compute_betweenness_and_closeness(agraph);
-    let mut nodes = Vec::with_capacity(agraph.len());
+    let mut graph = Graph::new();
+    for (n, node) in indices.iter().enumerate() {
+        for connection in node {
+            if *connection > n {
```

Comment:
a mild suggestion if you want to have fun with Rust:
maybe it could be done in an idiomatic way with `node.iter().filter().for_each()` or something (or not sure which method-combination would work) 😃

Reply:
Changed.
@kylegranger when you'll be back please look at that change in your code - I hope I've expressed the same what was your intention.

Future remark @kylegranger : I've made a `construct_graph` function to be used inside IPS to create graph with IP addrs. Your loop here does similar thing but constructing graph with node indexes. Maybe we'd create common solution for graph reconstruction or use one graph with some struct inside (having IP and index in it).